### DISCUSSION TITLE
Unable to launch TESTs on LINUX UBUNTU

### DISCUSSION THREAD

**Discussion by swathisuja** [redhat-developer/vscode-extension-tester#1450] [Q&A]
Hi,

I have the extester set up and works fine in windows OS. But in Linux Ubuntu 22.04
Node 18.20.0 and also 20.12.1
Yarn 1.22.19
When I run the tests, it fails at launching browser with error:

1) "before all" hook in "{root}"
  2) "after all" hook in "{root}"
 
  0 passing (243ms)
  2 failing
 
  1) "before all" hook in "{root}":
     SessionNotCreatedError: session not created: Chrome failed to start: exited normally.
  (session not created: DevToolsActivePort file doesn't exist)
  (The process started from chrome location /home/workspace/resources/VSCode-linux-x64/code is no longer running, so ChromeDriver is assuming that Chrome has crashed.)
      at Object.throwDecodedError (node_modules/selenium-webdriver/lib/error.js:521:15)
      at parseHttpResponse (node_modules/selenium-webdriver/lib/http.js:514:13)
      at Executor.execute (node_modules/selenium-webdriver/lib/http.js:446:28)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
 
  2) "after all" hook in "{root}":
     TypeError: Cannot read properties of undefined (reading 'manage')
      at VSBrowser.quit (node_modules/vscode-extension-tester/out/browser.js:174:44)
      at Context.<anonymous> (node_modules/vscode-extension-tester/out/suite/runner.js:126:31)
      at process.processImmediate (node:internal/timers:478:21)

Screenshot of Proper installation of Extension vsix, vscode, chrome driver :
![image](https://github.com/user-attachments/assets/f35c2cec-951a-49f8-b0ec-22c5ae42a165)

Supported chrome is installed in machine. Unable to run tests on Linux hence blocking from running of docker. 

Added comments in https://github.com/redhat-developer/vscode-extension-tester/issues/485 

Kindly look into this and help to get the tests run. 

Thanks,
Swathi.

**Comment by djelinek** [comment]
hello, are you running tests in some display? xvfb? frame or something else?

**Reply by swathisuja** [reply]
No @djelinek . A normal monitor with Linux processor. But monitor/machine is connected via VNC viewer. 

VS code did not launch. 
![image](https://github.com/user-attachments/assets/bd57e1c7-6fe7-4be8-8990-3b9aea0d1bcd)

**Reply by djelinek** [reply]
I don't know what exactly means "normal monitor with Linux processor", anyway I am asking to be sure the tests can be executed, which needs some monitor/display/frame.. so eg on our GH actions pipelines we are using xvfb (which simulates virtual display).. or eg if you now vncserver it is similar.. and than you are running tests in terminal using eg `DISPLAY=:<display_port> npm/yarn run <script>`

are you doing something similar to this?

**Reply by swathisuja** [reply]
yes, through vnc viewer remote connected to the physical machine with monitor. System has VS code into , open VS code , open the extension project and in fresh terminal trying to run "yarn run <script>

**Reply by djelinek** [reply]
I have noticed this warning in the log `test-mplab-extension-automation00.0.1-alpha: The engine "vscode" appears to be invalid.` what is the vscode engine defined in package.json of your extension?

**Reply by swathisuja** [reply]
sorry for late response.  
 "engines": {
        "vscode": "^1.75.0"
    },

if this is causing some sort of problem, we can update the version as per your suggestion

**Reply by djelinek** [reply]
sorry I was not available for a while.. it is really hard to investigate this way without knowing fully your env and setup of tests..

can you please try to change engine to at least eg `1.90.0` ?
